cpp/include/raft/stats/histogram.cuh

```diff
@@ -61,6 +62,44 @@ void histogram(HistType type,
   detail::histogram<DataT, IdxT, BinnerOp>(type, bins, nbins, data, nrows, ncols, stream, binner);
 }
 
+/**
+ * @brief Perform histogram on the input data. It chooses the right load size
+ * based on the input data vector length. It also supports large-bin cases
+ * using a specialized smem-based hashing technique.
+ * @tparam DataT input data type
+ * @tparam IdxType data type used to compute indices
+ * @tparam BinnerOp takes the input data and computes its bin index
+ * @tparam LayoutPolicy Layout type of the input data.
+ * @tparam AccessorPolicy Accessor for the input and output, must be valid accessor on
+ *                        device.
+ * @param handle the raft handle
+ * @param type histogram implementation type to choose
+ * @param bins the output bins (length = ncols * nbins)
+ * @param data input data (length = ncols * nrows)
+ * @param binner the operation that computes the bin index of the input data
+ *
+ * @note signature of BinnerOp is `int func(DataT, IdxT);`
+ */
+template <typename DataT,
+          typename IdxType  = int,
+          typename BinnerOp = IdentityBinner<DataT, IdxType>,
+          typename LayoutPolicy,
+          typename AccessorPolicy>
+void histogram(const raft::handle_t& handle,
+               HistType type,
+               raft::mdspan<int, raft::matrix_extent<IdxType>, LayoutPolicy, AccessorPolicy> bins,
```

Comment:
Given that `bins` is an mdspan of `int`, it looks like the implementation currently only knows how to deal with `int` indices.  Thus, would you consider either requiring that `IdxType` is `int`, or checking (possibly at run time) whether the arrays' extents fit in `int`?